Add override mistake (#137)
tc39/how-we-work: terminology.md

### References

[Import map proposal](https://github.com/WICG/import-maps#supplying-out-of-band-metadata-for-each-module)

> ## Override mistake

Reviewer: ```suggestion

## Override mistake
```

[Import map proposal](https://github.com/WICG/import-maps#supplying-out-of-band-metadata-for-each-module)

## Override mistake

### Definition

The behavior that ECMAScript throws a TypeError (in the strict mode) for code `O[K] = ...` when `K` is a non-writable property in the prototype of `O`.

> This behavior brings serious compatibility problems in the ecosystem when the runtime is "frozen" (which means all built-in APIs are frozen) for security or performance reasons.

Reviewer 2: I think we've had more compat problems with new properties than with frozen intrinsics, actually. Might as well mention both:

```suggestion
This behavior brings serious compatibility problems in the ecosystem when any properties are non-writable, including when [the language adds new built-in non-writable properties to existing prototypes](https://github.com/tc39/proposal-iterator-helpers/issues/115) and when [the runtime or code on the page freezes built-ins](https://github.com/tc39/proposal-iterator-helpers/issues/286).
```

Author: oh we're very unlucky, I don't know iterator helper hit both of them

### Example

```js
'use strict'
// environment
Object.freeze(Error.prototype)

// library code
function MyError() {
    var e = Error()
    e.name = 'MyError'
}
MyError.prototype.__proto__ = Error.prototype
// later

> new MyError() // TypeError: name is readonly

Author: I'm a bit unfamiliar with how we write class in ES5 days. the library code is an imaginary ES5 library

[Import map proposal](https://github.com/WICG/import-maps#supplying-out-of-band-metadata-for-each-module)

## Override mistake

### Definition

> [There is no universal consensus that this is a mistake](https://github.com/tc39/ecma262/pull/1307#issuecomment-421094561).

Reviewer 3: ```suggestion
[There is no consensus that this is a mistake](https://github.com/tc39/ecma262/pull/1307#issuecomment-421094561).
```

nit: consensus is universal by definition, no?
